File: android/src/main/java/org/ligi/passandroid/ui/pass_view_holder/VerbosePassViewHolder.java
package org.ligi.passandroid.ui.pass_view_holder;

import android.app.Activity;
import android.view.View;
import org.ligi.passandroid.model.PassStore;
import org.ligi.passandroid.model.pass.Pass;

public class VerbosePassViewHolder extends PassViewHolder {

    public VerbosePassViewHolder(View view) {
        super(view);
    }

    public void apply(final Pass pass, final PassStore passStore, final Activity activity) {
        super.apply(pass, passStore, activity);

        String stringToSetForDateOrExtraText = null;

        if (getTimeInfoString() != null) {
            stringToSetForDateOrExtraText = getTimeInfoString();
        } else if (getExtraString() != null) {
            stringToSetForDateOrExtraText = getExtraString();
        }

        if (stringToSetForDateOrExtraText != null) {
            dateOrExtraText.setText(stringToSetForDateOrExtraText);
            dateOrExtraText.setVisibility(View.VISIBLE);
        } else {
            dateOrExtraText.setVisibility(View.GONE);
        }
    }
}

File: android/src/main/java/org/ligi/passandroid/ui/pass_view_holder/VerbosePassViewHolder.kt
package org.ligi.passandroid.ui.pass_view_holder

import android.app.Activity
import androidx.cardview.widget.CardView
import android.view.View
import android.widget.TextView
import org.ligi.passandroid.R
import org.ligi.passandroid.model.PassStore
import org.ligi.passandroid.model.pass.Pass

open class VerbosePassViewHolder(view: CardView) : PassViewHolder(view) {

    override fun apply(pass: Pass, passStore: PassStore, activity: Activity) {
        super.apply(pass, passStore, activity)

        val dateOrExtraText = getTimeInfoString(pass) ?: getExtraString(pass)

        val date = view.findViewById<TextView>(R.id.date)
        if (dateOrExtraText != null && dateOrExtraText.isNotEmpty()) {
            date.text = dateOrExtraText
            date.visibility = View.VISIBLE
        } else {
            date.visibility = View.GONE
        }
    }
}
